torchgeo/datasets/seasonet.py

```diff
@@ -0,0 +1,456 @@
+        sample = {"image": image, "mask": mask}
+
+        if self.transforms is not None:
+            sample = self.transforms(sample)
+
+        return sample
+
+    def __len__(self) -> int:
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    def _load_image(self, index: int) -> Tensor:
+        """Load image(s) for a single location.
+
+        Args:
+            index: index to return
+
+        Returns:
+            the stacked seasonal images
+        """
+        paths = self.files.iloc[index]
+        if self.concat_seasons > 1:
+            paths = random.sample(paths, self.concat_seasons)
+        tensor = torch.empty(self.concat_seasons * self.channels, *self.image_size)
+        for img_idx, path in enumerate(paths):
+            bnd_idx = 0
+            for band in self.bands:
+                with rasterio.open(f"{path}_{band}.tif") as f:
+                    array = f.read(
+                        out_shape=[f.count] + list(self.image_size),
+                        out_dtype="int32",
+                        resampling=Resampling.bilinear,
+                    )
+                image = torch.from_numpy(array).float()
+                c = img_idx * self.channels + bnd_idx
+                tensor[c : c + image.shape[0]] = image
+                bnd_idx += image.shape[0]
+        return tensor
+
+    def _load_target(self, index: int) -> Tensor:
+        """Load the target mask for a single location.
+
+        Args:
+            index: index to return
+
+        Returns:
+            the target mask
+        """
+        path = self.files.iloc[index][0]
+        with rasterio.open(f"{path}_labels.tif") as f:
+            array = f.read() - 1
+        tensor = torch.from_numpy(array).squeeze().long()
+        return tensor
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        # Check if all files already exist
+        if all(
+            os.path.exists(os.path.join(self.root, name))
+            for name in self.extracted_names
+        ):
+            return
+
+        # Check for downloaded files and extract existing zips
+        missing = []
+        for file_name in self.file_names:
+            file_path = os.path.join(self.root, file_name)
+            if not os.path.exists(file_path):
+                missing.append(file_name)
+            elif file_name[-3:] == "zip":
+                extract_archive(file_path)
+
+        if not missing:
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            missing_str = (
+                f"{', '.join(missing[:-1])} and {missing[-1]}"
+                if len(missing) > 1
+                else missing[0]
+            )
```

Comment:
Not sure if the error message needs to be this detailed

Reply:
True, it doesn't have to be. At least the extra logic just to add an "and". I do think it is helpful to print the actual list of missing files though, as the dataset requires multiple very large files to be downloaded and just saying "dataset missing", even when it's just a single file, may be a bit misleading.